File: src/nodejs/redis_om_search_example/load_data.js
import { Client } from 'redis-om';
import { bookSchema } from './model.js';
import { readdir, readFile } from 'node:fs/promises';
import { sep as pathSeparator } from 'node:path';

const FILE_PATH = '../../../data/books';

const client = new Client();
await client.open(process.env.REDIS_OM_URL || 'redis://localhost:6379');

const bookRepository = client.fetchRepository(bookSchema);

const fileNames = await readdir(FILE_PATH);
for (const fileName of fileNames) {
  console.log(`${FILE_PATH}${pathSeparator}${fileName}`);
  const bookData = JSON.parse(await readFile(`${FILE_PATH}${pathSeparator}${fileName}`));
  console.log(bookData);
}

await client.close();

File: src/python/redis_om_search_example/load_data.py
import io
import json
import os

from model import Book
from redis_om import Migrator

# Load the sample data set into Book instances and save to Redis
for filename in os.listdir("../../../data/books"):
    f = os.path.join("../../../data/books", filename)

    if os.path.isfile(f):
        book_file = io.open(f, encoding="utf-8")
        book_json = json.load(book_file)
        book_file.close()

        book = Book(**book_json)
        book.save()
        print(f"Stored book {book.title}.")
 
# Create the search index.
Migrator().run()
print("Created search index.")
